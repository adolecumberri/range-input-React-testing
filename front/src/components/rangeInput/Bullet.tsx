import React, { CSSProperties, FC } from 'react';

interface IBullet {
    size: number;
    position: number;
    [x: string]: any;

}

const Bullet: FC<IBullet> = ({ size, position, sliderMargin, setSelectedBullet, type }) => {

    const handleDragEnter = (e: any) => {
        e.preventDefault();
        e.stopPropagation();
        setSelectedBullet(type);

    };

    const bulletStyle: CSSProperties = {
        border: "1px solid black",
        boxSizing: "border-box",
        position: "absolute",
    }

    return (<>
        <div
        data-testid={`bullet-${type}`}
            className="range-bullet range-bullet-hidden"
            style={{
                ...bulletStyle,
                width: size,
                height: size,
                marginLeft: position - (size / 2),
                opacity: 0,
                zIndex: 3,
            }}
            draggable
            onDragEnter={e => handleDragEnter(e)}
        >
        </div>

        <div
            className="range-bullet range-bullet-visible"
            style={{
                ...bulletStyle,
                width: size,
                height: size,
                marginLeft: position - (size / 2),
                backgroundColor: "white",
                zIndex: 2
            }}
            draggable
        >
        </div>

    </>
    );
}

export default Bullet;
