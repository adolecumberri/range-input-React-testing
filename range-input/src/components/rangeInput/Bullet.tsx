import React, { CSSProperties, FC, useState } from 'react';



interface IBullet {
    size: number;
    position: number;
    [x: string]: any;

}

const Bullet: FC<IBullet> = ({ size, position, sliderMargin, setSelectedBullet, type }) => {

    const handleDragEnter = (e: any) => {
        console.log("enter");
        e.preventDefault();
        e.stopPropagation();
        setSelectedBullet(type);

    };

    const handleDragLeave = (e:any) => {
        e.preventDefault();
        e.stopPropagation();
    }

    const bulletStyle: CSSProperties = {
        border: "1px solid black",
        boxSizing: "border-box",
        position: "absolute",
    }

    return (<>
        <div 
        className="range-bullet range-bullet-hidden"
        style={{
            ...bulletStyle,
            width: size,
            height: size,
            marginLeft: position - (size / 2),
            opacity: 0,
            zIndex: 3,
        }}
            draggable
            onDragEnter={e => handleDragEnter(e)}
            onDragLeave={e => handleDragLeave(e)}
        >
        </div>

        <div 
        className="range-bullet range-bullet-visible"
        style={{
            ...bulletStyle,
            width: size,
            height: size,
            marginLeft: position - (size / 2),
            backgroundColor: "white",
            zIndex: 2
        }}
            draggable
        >
        </div>

    </>
    );
}

export default Bullet;
