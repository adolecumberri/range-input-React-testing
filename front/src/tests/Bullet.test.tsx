import React from "react";
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from "@testing-library/react";
import Bullet from '../components/rangeInput/Bullet';

import renderer from 'react-test-renderer';

describe('Bullet general test.', () => {
    it("renders correctly", () => {

        render(<Bullet
            position={0}
            setSelectedBullet={()=>{}}
            size={40}
            sliderMargin={0}
            type={"min"}

        />);

        expect(screen.getByTestId("bullet-min")).toBeTruthy();
    });

    it("Bullet changes when hovered", () => {
        const component = renderer.create(<Bullet
            position={0}
            setSelectedBullet={()=>{}}
            size={40}
            sliderMargin={0}
            type={"min"}
        />);

        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

         // manually trigger the callback
         const data = {
           preventDefault: () => {},
           stopPropagation: () => {},
         };
        (tree as renderer.ReactTestRendererJSON[])[0].props.onDragEnter(data);

         // re-rendering
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
        });
});

