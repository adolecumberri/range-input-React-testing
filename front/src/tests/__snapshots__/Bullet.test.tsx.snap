// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bullet general test. Bullet changes when hovered 1`] = `
Array [
  <div
    className="range-bullet range-bullet-hidden"
    data-testid="bullet-min"
    draggable={true}
    onDragEnter={[Function]}
    style={
      Object {
        "border": "1px solid black",
        "boxSizing": "border-box",
        "height": 40,
        "marginLeft": -20,
        "opacity": 0,
        "position": "absolute",
        "width": 40,
        "zIndex": 3,
      }
    }
  />,
  <div
    className="range-bullet range-bullet-visible"
    draggable={true}
    style={
      Object {
        "backgroundColor": "white",
        "border": "1px solid black",
        "boxSizing": "border-box",
        "height": 40,
        "marginLeft": -20,
        "position": "absolute",
        "width": 40,
        "zIndex": 2,
      }
    }
  />,
]
`;

exports[`Bullet general test. Bullet changes when hovered 2`] = `
Array [
  <div
    className="range-bullet range-bullet-hidden"
    data-testid="bullet-min"
    draggable={true}
    onDragEnter={[Function]}
    style={
      Object {
        "border": "1px solid black",
        "boxSizing": "border-box",
        "height": 40,
        "marginLeft": -20,
        "opacity": 0,
        "position": "absolute",
        "width": 40,
        "zIndex": 3,
      }
    }
  />,
  <div
    className="range-bullet range-bullet-visible"
    draggable={true}
    style={
      Object {
        "backgroundColor": "white",
        "border": "1px solid black",
        "boxSizing": "border-box",
        "height": 40,
        "marginLeft": -20,
        "position": "absolute",
        "width": 40,
        "zIndex": 2,
      }
    }
  />,
]
`;
